src/com/mambu/apisdk/util/RequestExecutorImpl.java

```diff
@@ -165,6 +141,284 @@ public String executeRequest(String urlString, ParamsMap params, Method method,
 		return response;
 	}
 
+	/*
+	 * Executes a request in order to download content and returns it as a ByteArrayOutputStream
+	 */
+	@Override
+	public ByteArrayOutputStream executeRequest(String urlString, ParamsMap params, ApiDefinition apiDefinition)
+			throws MambuApiException {
```

Comment:
Yes, this method might work for other API return types (zip, jar, doc). I`m not sure I can get your point regarding the second part. Initially I was thinking about something like:
if(!apiDefinition.getApiReturnFormat().equals(ApiReturnFormat.ZIP_ARCHIVE)){
        throw new UnsupportedOperationException();
}
But we already do this kind of check in the ServiceExecutor line 166. That is how we make sure this method is called only fie ZIP_ARCHIVE return type.